js/ui/overlay/ui.overlay.js

```diff
@@ -502,12 +472,44 @@ const Overlay = Widget.inherit({
 
     _normalizePosition: function() {
         const position = this.option('position');
-        this._position = typeof position === 'function' ? position() : position;
+        const defaultPositionOptions = {
+            of: this.option('target'),
+            boundaryOffset: DEFAULT_BOUNDARY_OFFSET
+        };
+        if(isFunction(position)) {
+            this._position = extend(true, {}, defaultPositionOptions, position());
```

Comment:
function passed to `position` property can return alias string. Seems like this case is broken.

Let's use `getOptionValue`?

Reply:
reworked